### PR TITLE
Implement two-handed weapons #229

### LINKED ISSUE #229: Implement two-handed weapons
For example bow should be two-handed. So if you wear the bow you can't wear the shield at the same time.

### PR DESCRIPTION
**Pull request by neu5**

### CODE REVIEW

File: server/gameObjects/Player.mjs [hunk 1]
```diff
@@ -122,8 +101,36 @@ export default class Player {
...
   }
 
-  canAddToBackpack(itemsToAddNum) {
-    return itemsToAddNum <= this.backpack.slots - this.backpack.items.length;
+  setOnline(socketId) {
+    this.isOnline = true;
+    this.socketId = socketId;
+  }
+
+  setSelectedObject(player) {
```
Comment (przenz) [inline server/gameObjects/Player.mjs:110]: `setSelectedPlayer`?

File: server/gameObjects/Player.mjs [hunk 2]
```diff
@@ -238,20 +292,22 @@ export default class Player {
...
-
-      this.destroyItemFromBackpack(itemName);
-      this.equipment.backpack = item;
+      this.moveBackpackToEquipment(itemName);
+    }
+    // player wants to wear two-handed weapon and wears the shield
+    else if (itemSchema.isTwoHanded && this.equipment.shield !== undefined) {
+      if (!this.moveTwoHandedWeaponToEquipment(itemName)) {
+        return false;
+      }
```
Comment (przenz) [inline server/gameObjects/Player.mjs]:
```suggestion
      return this.moveTwoHandedWeaponToEquipment(itemName);
```

File: server/gameObjects/Player.mjs [hunk 3]
```diff
@@ -238,20 +292,22 @@ export default class Player {
...
+    }
+    // player wants to wear shield
+    else if (
+      itemSchema.type === ITEM_TYPES.SHIELD &&
+      this.hasTwoHandedWeapon()
+    ) {
+      if (!this.moveShieldToEquipment(itemName)) {
+        return false;
+      }
```
Comment (przenz) [inline server/gameObjects/Player.mjs]:
```suggestion
      return this.moveShieldToEquipment(itemName);
```

**Review by przenz — CHANGES_REQUESTED** [review]

File: server/gameObjects/Player.mjs (hunk 1 above)
Comment (neu5) [reply server/gameObjects/Player.mjs:110]: internally it should be `this.selectedObject` or something like that, but let's deal with it some time in the future

**Review by przenz — APPROVED** [review]

**Pull request merged**